src/lib/fragments/Accordion.react.js

```diff
@@ -4,42 +4,207 @@ import xtype from 'xtypejs';
+            const contentAtPreviousIndex = this.contentRefs[previousIndex].current;
+            if (contentAtPreviousIndex) {
+              contentAtPreviousIndex.focus();
+              return;
+            }
+          } else { // previous section is closed so focus the previous heading
+            newIndex = index - 1;
+          }
+        }
+        break;
+      case EventOrigin.SECTION_CONTENT: // focus the heading at the same level as the content
+        newIndex = index;
+        break;
+      default:
+        break;
+    }
 
+    // focus the heading at level newIndex
+    const headingToFocus = this.headerRefs[newIndex].current;
+    if (headingToFocus) {
+      headingToFocus.focus();
+    }
+  }
+
+  // This method handles the Down arrow key event for an accordion component
+  // depending on the origin of the event (Show All button, section heading, or section content).
+  handleDownArrow = (index, eventOrigin) => {
+    const numberSections = this.props.accordionHeadings.length;
+    let newIndex = index;
+
+    switch (eventOrigin) {
+      case EventOrigin.SHOW_ALL_BUTTON:
+        newIndex = 0; // go to accordion heading 0
+        break;
+      case EventOrigin.SECTION_HEADING:
+        const sectionIsOpen = this.state.sectionsOpen[index];
+        if (sectionIsOpen) {
+          const content = this.contentRefs[index].current;
+          if (content) {
+            content.focus();
+            return;
+          }
+        } else { // section is closed, go to next heading if there is one
+          newIndex = index + 1;
+        }
+        break;
+      case EventOrigin.SECTION_CONTENT:
+        newIndex = index + 1; // go to next heading if there is one
+        break;
+      default:
+        break;
+    }
+
+    if (newIndex >= 0 && newIndex < numberSections) {
+      const nextHeading = this.headerRefs[newIndex].current;
+      if (nextHeading) {
+        nextHeading.focus();
+        return;
+      }
+    } else if (newIndex >= numberSections) { // focus on the next element on the page
+      const currentHeading = this.headerRefs[index].current;
+
+      const nextElement = this.findFocusableElement(currentHeading, "next");
+      if (nextElement) {
+        nextElement.focus();
+      }
+    }
+  }
+
+  handleKeyEvent = (event, index) => {
+    let eventOrigin;
+
+    if (index === -1) {
+      eventOrigin = EventOrigin.SHOW_ALL_BUTTON;
+    } else if (event.target.className === AccordionContentClassName) {
+      eventOrigin = EventOrigin.SECTION_CONTENT;
+    } else if (event.target.className === AccordionButtonClassName) {
+      eventOrigin = EventOrigin.SECTION_HEADING;
+    } else {
+      return;
+    }
+
+    switch (event.key) {
+      case 'ArrowDown':
+        event.preventDefault();
+        this.handleDownArrow(index, eventOrigin);
+        break;
+      case 'ArrowUp':
+        event.preventDefault();
+        this.handleUpArrow(index, eventOrigin);
+        break;
+      default:
+        break;
```

Comment:
Without firing this up does this support tabs. W3 standards for accordion keyboard navigation 
![image](https://user-images.githubusercontent.com/2092344/234830245-1f8d7f04-d21a-476b-ad76-0a913fbaa021.png)

GovUK also doesn't using arrows to navigate only tab and enter to go intoit

Reply:
I don't think it does, I will sort that out

Reply:
The tab index on the accordion was set to -1, this was causing the content to not be focused on my pressing tab so I have changed it to 0 which added it back to the natural keyboard navigation order

Reply:
I have just realised that the down and up arrows are going onto the content,  when it seems like it should be going to the headers, will fix

Reply:
Thank you, you will also need the enter key because it shouldnt just go to it from my understanding

Reply:
Do you mean for expanding and collapsing, it already does that, I believe it's the default behaviour

Reply:
When you use arrows or tabs on unopened it shouldn't open it, it should read and let you decide to open. If its opened then it should read and allow you to close. (Should be same with arrows if im not mistaken

Reply:
I have used the w3 guidelines, enter or spacebar opens and closes the accordion, and tab moves onto the next focusable element. I have removed all code from the 'handle-up-and-down buttons' functions that refer to any 'accordion-content' as the up-and-down buttons only want to go from header to header.

Reply:
Amazing good work :) 

I always like to keep an eye out on this repo and make suggestions in case people wasn't aware of them

Reply:
Thank you, your comments were very helpful